src/types.rs

```diff
@@ -344,6 +344,15 @@ impl<T> Line<T>
     }
 }
 
+impl<T> PartialEq for Line<T>
+    where T: Float
+{
+    fn eq(&self, other: &Self) -> bool {
+        (self.start == other.start && self.end == other.end) ||
+            (self.start == other.end && self.end == other.start)
```

Comment:
hmmm. thinking out loud, i'm wondering if we should avoid `start` and `end` as terms for `Line`. 'start' and 'end' seem to imply that a `Line` is directed, even though it's not.

Reply:
I agree. `a` and `b`?